src/math/src/signed_u256.cairo

```diff
@@ -225,36 +124,192 @@ impl i257PartialOrd of PartialOrd<i257> {
 
     fn gt(lhs: i257, rhs: i257) -> bool {
         // Check if `lhs` is negative and `rhs` is positive.
-        if (lhs.sign & !rhs.sign) {
+        if lhs.is_negative & !rhs.is_negative {
             return false;
         }
         // Check if `lhs` is positive and `rhs` is negative.
-        if (!lhs.sign & rhs.sign) {
+        if !lhs.is_negative & rhs.is_negative {
             return true;
         }
         // If `lhs` and `rhs` have the same sign, compare their absolute values.
-        if (lhs.sign & rhs.sign) {
-            return lhs.inner < rhs.inner;
+        if lhs.is_negative & rhs.is_negative {
+            lhs.abs < rhs.abs
         } else {
-            return lhs.inner > rhs.inner;
+            lhs.abs > rhs.abs
         }
     }
 }
 
+
+// Divides the first i257 by the second i257.
+
 // Implements the Neg trait for i257.
 impl i257Neg of Neg<i257> {
     fn neg(a: i257) -> i257 {
-        i257 { inner: a.inner, sign: !a.sign }
+        i257_neg(a)
     }
 }
 
+impl i257Zeroable of Zeroable<i257> {
+    fn zero() -> i257 {
+        i257_new(0, false)
+    }
+    fn is_zero(self: i257) -> bool {
+        self == Zeroable::zero()
+    }
+    fn is_non_zero(self: i257) -> bool {
+        self != Zeroable::zero()
+    }
+}
+
+// Checks if the given i257 integer is zero and has the correct sign.
+// # Arguments
+// * `x` - The i257 integer to check.
+// # Panics
+// Panics if `x` is zero and is not negative
+fn i257_assert_no_negative_zero(x: i257) {
+    if x.abs == 0 {
+        assert(!x.is_negative, 'negative zero');
+    }
+}
+
+// Adds two i257 integers.
+// # Arguments
+// * `a` - The first i257 to add.
+// * `b` - The second i257 to add.
+// # Returns
+// * `i257` - The sum of `a` and `b`.
+fn i257_add(lhs: i257, rhs: i257) -> i257 {
```

Comment:
why did you move all this functions outside of the traits ?

Reply:
to have a cleaner diff with `i128` which is also used in some projects

Reply:
or at least, the corelib implementation wasn't sufficient enough and that's why it needed to be redone in satoru iirc

Reply:
keep them in the traits plz

Reply:
done